### DISCUSSION TITLE
Mandatory fields of ui.table

### DISCUSSION THREAD

**Discussion by denniswittich** [zauberzeug/nicegui#3729] [Ideas / Feature Requests]
### Description

When creating a ui.table with columns one needs to provide `name`, `label` and `field`. This seemed a bit redundant first but after some discussions I get that `field` provides some extra functionalities that can be useful. However, when working with tables for the first time one could forget to set `field` in the columns dict. This modification of the web example

```python
from nicegui import ui

columns = [
    {'name': 'name', 'label': 'Name', 'required': True, 'align': 'left'},
    {'name': 'age', 'label': 'Age', 'sortable': True},
]
rows = [
    {'name': 'Alice', 'age': 18},
    {'name': 'Bob', 'age': 21},
    {'name': 'Carol'},
]
ui.table(columns=columns, rows=rows, row_key='name')

ui.run()
```   

does not produce an error but instead a table where the number of lines matches but without any content:

![image](https://github.com/user-attachments/assets/1a86cb53-f66e-468c-a854-afb75e759285)

My suggestion is to validate the dictionary and then either default to the `name` if `field` is missing or to throw an error.

**Comment by rodja** [comment]
I like the idea of allowing to omit certain fields in the rows dicts. We could also infer the label from name (similar as [omitting the whole rows parameter](https://nicegui.io/documentation/table#omitting_columns)):

```py
    columns = [
        {'name': 'name', 'align': 'left'},
        {'name': 'age', 'sortable': True},
    ]
    rows = [
        {'name': 'Alice', 'age': 18},
        {'name': 'Bob', 'age': 21},
        {'name': 'Carol'},
    ]
    ui.table(columns=columns, rows=rows, row_key='name')
    ui.table(columns=columns, rows=rows,  row_key='name')
```

Or the name from label:

```py
    columns = [
        {'label': 'Name', 'align': 'left'},
        {'label': 'Age', 'sortable': True},
    ]
    rows = [
        {'Name': 'Alice', 'Age': 18},
        {'Name': 'Bob', 'Age': 21},
        {'Name': 'Carol'},
    ]
    ui.table(columns=columns, rows=rows, row_key='Name')
    ui.table(columns=columns, rows=rows,  row_key='Name')
```
